Updates PSU Update

JPRuskin opened the pull request:
This should now source v4.x for the package.

- Grabs list of MSI links from the all-releases page, sorts them by version (ignoring prereleases), uses the top one

Fixes #4 

Also noticed a warning and error in the GHA runs, so tried to fix those up!

- Updated $Env: names to match the ones set in the Update.yml
- Updated actions/checkout to v4, as v2 was warning of deprecation

Didn't add anything to make the PowerShellUniversal.Licensed package build, but that could either be triggered as an extra GHA step or by adding something like `$PSScriptRoot\UniversalLicensed.ps1 -DashboardVersion $DashboardVersion` to the end of the `Universal.ps1` script.

JPRuskin committed:
Fixes the PSU Update Script for >3.x

The update script was using an API endpoint that seems to be stuck reporting 2.x. As 3.x is now feature complete, and 4.x is in full swing, this is unfortunate!

This changes the update script to source versions from the all-release downloads page, and to use environment variable names as shown in the Update GHA.

JPRuskin commented:
@steviecoaster Any thoughts on this?

JPRuskin commented:
@steviecoaster It would be great to get this merged! Let me know if there's anything else I can do, here.

Had a quick go, and the logic still seems good.

The only additional thing we may want to consider is that I've found my instance refuses to start (similar to [this](https://support.ironmansoftware.com/portal/en/kb/articles/powershell-universal-service-crashes-on-startup-after-an-upgrade-to-1-4-6)) twice after upgrading straight from CurrentPackageVersion to 4.x. I think it's down to the database change. I've solved it in both cases by deleting the old database, but that's a bit harsh to inflict - [this](https://forums.ironmansoftware.com/t/upgrade-from-1-5-13-to-2-1-0-now-failing-to-start/5183/4) may be a better solution?

steviecoaster commented:
> @steviecoaster It would be great to get this merged! Let me know if there's anything else I can do, here.
> 
> Had a quick go, and the logic still seems good.
> 
> The only additional thing we may want to consider is that I've found my instance refuses to start (similar to [this](https://support.ironmansoftware.com/portal/en/kb/articles/powershell-universal-service-crashes-on-startup-after-an-upgrade-to-1-4-6)) twice after upgrading straight from CurrentPackageVersion to 4.x. I think it's down to the database change. I've solved it in both cases by deleting the old database, but that's a bit harsh to inflict - [this](https://forums.ironmansoftware.com/t/upgrade-from-1-5-13-to-2-1-0-now-failing-to-start/5183/4) may be a better solution?

Hard to do this _correctly_ in a beforeModify script, but putting it early in the install script should be a simple enough fix. Let's add it before we merge this.

JPRuskin committed:
Adds Database Upgrade to PSU Install / BeforeModify

If the connectionstring doesn't contain upgrade=true, we add it (if the user doesn't specify otherwise).

JPRuskin committed:
Convert PSU to AU
